src/dev_server.rs

```diff
@@ -4,31 +4,73 @@ use clap::Parser;
 use std::{
     fs,
     io::{prelude::*, BufReader},
-    net::{IpAddr, SocketAddr, TcpListener, TcpStream},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener, TcpStream},
     path::Path,
     process,
 };
 
+/// A simple HTTP server that can, optionally, watch a given command.
+///
+/// Get the files at `watch_path` and serve them at a given IP address
+/// (127.0.0.1:8000 by default). An optional command can be run along the
+/// server, relaunched if changes are detected in your project via a [`Watch`].
+///
+/// # Usage
+///
+/// ```no_run
+/// use xtask_wasm::{metadata, DevServer};
+/// use std::process::Command;
+///
+/// let build_path = metadata().workspace_root.join("build");
+/// let mut command = Command::new("cargo");
+/// command.arg("run");
+///
+/// DevServer::new()
+/// .command(command)
+/// .start(build_path)
+/// .expect("cannot run dev server process");
```

Comment:
I know indenting in the doc comments is hard.

There are 3x `/`, followed by a space, then if you need to indent code you need to add 4 more spaces.

```suggestion
///     .command(command)
///     .start(build_path)
///     .expect("cannot run dev server process");
```